runtime.md

```diff
@@ -67,28 +65,43 @@ This operation MUST return the state of a container as specified in the [State](
 In particular, the state MUST be serialized as JSON.
 
 
-### Start
+### Create
 
-`start <container-id> <path-to-bundle>`
+`create <container-id> <path-to-bundle>`
 
 This operation MUST generate an error if it is not provided a path to the bundle and the container ID to associate with the container.
-If the ID provided is not unique across all containers within the scope of the runtime, or is not valid in any other way, the implementation MUST generate an error.
-Using the data in `config.json`, that are in the bundle's directory, this operation MUST create a new container.
-This includes creating the relevant namespaces, resource limits, etc and configuring the appropriate capabilities for the container.
-A new process within the scope of the container MUST be created as specified by the `config.json` file otherwise an error MUST be generated.
+If the ID provided is not unique across all containers within the scope of the runtime, or is not valid in any other way, the implementation MUST generate an error and a new container MUST not be created.
+Using the data in [`config.json`](config.md), that is in the root of the bundle's directory, this operation MUST create a new container.
+This includes creating, or entering, the namespaces specified in the [`config.json`](config.md), resource limits, etc and configuring the appropriate capabilities for the container.
+If the `config.json` specifies that a PID namespace is to be created then one MUST be created, but the user-specified code within that namespace MUST NOT be created at this time.
+In some implementations this means that a temporary process is created in the PID namespace but it pauses until the `start` operation is invoked before replacing the process with the user-specified code.
+
+### Start
+`start <container-id>`
 
-Attempting to start an already running container MUST have no effect on the container and MUST generate an error.
+This operation MUST generate an error if it is not provided the container ID.
+This operation MUST create, and start, the user-specified code as specified by the [`process`](config.md#process-configuration) file otherwise an error MUST be generated and no process MUST be started.
+This process MUST be run in the PID namespace associated with the container.
 
-### Stop
+Attempting to start an already started container MUST have no effect on the container and MUST generate an error.
 
-`stop <container-id>`
+### Run
+`run <container-id>`
 
 This operation MUST generate an error if it is not provided the container ID.
+This operation MUST invoke the `create` operation, and if there are no errors, followed by the `start` operation.
```

Comment:
I like that - thanks - updated
-Doug

> On Apr 14, 2016, at 9:59 AM, Julian Friedman <notifications@github.com> wrote:
> 
> In runtime.md <https://github.com/opencontainers/runtime-spec/pull/384#discussion_r59683493>:
> 
> >  
> >  This operation MUST generate an error if it is not provided the container ID.
> > +This operation MUST invoke the `create` operation, and if there are no errors, followed by the `start` operation.
> Maybe "MUST invoke the create operation and then immediately start the process named in config.json. The runtime is not required to pause between creating the container and executing the requested process." (To clarify the intent that run need not actually perform the socket/cgroup/other-implementation steps implied by create/start)
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub <https://github.com/opencontainers/runtime-spec/pull/384/files/6d80e4687703d9dae5f669b48b5072f61dc3cd9f#r59683493>